### PR TITLE
MdePkg: Add http error 429 support

### PR DESCRIPTION
**Pull request by kenlautner**
# Description

Add HTTP Error code 429.  HTTP status codes have to be explicitly specified in DxeHttpLib in order to be supported. Intune may return 429 to throttle requests.

<_For each item, place an "x" in between `[` and `]` if true. Example: `[x]` (you can also check items in GitHub UI)_>

<_Create the PR as a Draft PR if it is only created to run CI checks._>

<_Delete lines in \<\> tags before creating the PR._>

- [ ] Breaking change?
  - **Breaking change** - Does this PR cause a break in build or boot behavior?
  - Examples: Does it add a new library class or move a module to a different repo.
- [ ] Impacts security?
  - **Security** - Does this PR have a direct security impact?
  - Examples: Crypto algorithm change or buffer overflow fix.
- [ ] Includes tests?
  - **Tests** - Does this PR include any explicit test code?
  - Examples: Unit tests or integration tests.

## How This Was Tested

Tested that Qemu is able to compile with the change.

## Integration Instructions

N/A

### CODE REVIEW

File: NetworkPkg/Library/DxeHttpLib/DxeHttpLib.c [hunk 1]
```diff
@@ -2077,6 +2077,8 @@ HttpMappingToStatusCode (
       return HTTP_STATUS_416_REQUESTED_RANGE_NOT_SATISFIED;
     case 417:
       return HTTP_STATUS_417_EXPECTATION_FAILED;
+    case 429:
```
Comment (mdkinney) [inline NetworkPkg/Library/DxeHttpLib/DxeHttpLib.c:2080]: Where to these constant values such as 429 come from?  Is there are include file for the related HTTP specs or should one be added?

Comment (mdkinney) [reply NetworkPkg/Library/DxeHttpLib/DxeHttpLib.c:2080]: Is this the official registry of these value?

https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml

Comment (kenlautner) [reply NetworkPkg/Library/DxeHttpLib/DxeHttpLib.c:2080]: That's correct

Comment (mdkinney) [reply NetworkPkg/Library/DxeHttpLib/DxeHttpLib.c:2080]: Can we add a reference to this registry in the function header.

Comment (kenlautner) [reply NetworkPkg/Library/DxeHttpLib/DxeHttpLib.c:2080]: Sure.  I'll add a link to the official http status codes there.

**Review by mdkinney — APPROVED** [review]

**Review by SaloniKasbekar — APPROVED** [review]

**Pull request merged**